grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java

```diff
@@ -238,12 +240,15 @@ static void generateConceptState(Representation resource, Concept concept){
                     .withProperty(BASETYPE_PROPERTY, getBaseType(instance).name());
         } else {
             Type type = concept.asType();
-            resource.withProperty(ID_PROPERTY, type.getName().getValue())
+            resource.withProperty(ID_PROPERTY, type.getId().getValue())
```

Comment:
You set the `ID_PROPERTY ` in the if(...) and the else(....). Could clean that up.